### PR TITLE
add MindPlaySound to role system

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
made tator and ninja sound use same thing
more stuff can use it in the future (malf, nukies but its a bit sus so idk)

## Why / Balance
good

## Technical details
added `MindPlaySound` to shared role system

## Media
it works trust
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by Bixkitts** [comment]
This is better than the random sprinkling of _audioSystem.PlayGlobal() throughout the code as it exists now,
but this does not need to be decoupled from the RoleSystem and RoleComponents in it's own system and component. 
I bet we can just notify the players from the existing MindAddRole() function and RoleComponents :v

**Comment by deltanedas** [comment]
i dont think adding a role should require a greeting sound, or that role system should even care about greetings. this way its independent of it

**Comment by Bixkitts** [comment]
That's also fair, this makes sense.
I was however under the impression that every role change in the game almost definitely notifies the player with message and/or sound and should do in almost all cases.
We could just put sound files and messages in the RoleComponent passed to MindAddRole() and have a notify function in
SharedRoleSystem that makes the player aware of their role IF the sound/message is non-null in their role component (all of which already exist for each role).
More components and systems is more clutter for ppl adding roles in the future, and besides, you didn't patch all the
roles to use this and I might :trollface:
I'll have a crack at my thing tomorrow and link the PR if it doesn't suck.

**Comment by deltanedas** [comment]
yeah idk about nukie or zombie since theyre not as clear cut

**Comment by Bixkitts** [comment]
Yeah I was scratching my head looking at the nukie code for a minute. 
They run a notify function with sound+message on round start instead of on role assignment 
like the traitor for example :shrug:

**Comment by deltanedas** [comment]
yeah thats why i left it out its a bit sus

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs [hunk 1]
```diff
@@ -268,21 +268,19 @@ public bool MakeTraitor(ICommonSession traitor, bool giveUplink = true, bool giv
...
+        // Assign briefing and greeting sound
         _roleSystem.MindAddRole(mindId, new RoleBriefingComponent
         {
             Briefing = briefing
-        });
+        }, mind);
+        _roleSystem.MindAddRole(mindId, new RoleSoundComponent
+        {
+            Sound = traitorRule.GreetSoundNotification
+        }, mind);
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:280]: Should probably put these in some helper method or something at some point.

Comment (deltanedas) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:280]: like for adding array of components or just for greeting

Comment (deltanedas) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:280]: @metalgearsloth pro

**Comment by Bixkitts** [comment]
So far this is literally just boilerplate to call PlayGlobal()

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:280]: The greeting

**Pull request merged**